### ISSUE TITLE
Layout shift causing click to miss target element

### ISSUE THREAD

**Issue by MadLittleMods** [cypress-io/cypress#5662]
### Current behavior:

There is a `blur` event handler which changes the height of an element and shifts the layout. It seems like when the click is made, it sends out a blur first which changes the layout and when the DOM-level click happens, it misses the element target.

![](https://i.imgur.com/Ped1hy1.gif)

You can see this in a real life test situation here, https://gitlab.com/gitlab-org/gitter/webapp/merge_requests/1642

![](https://i.imgur.com/KVJBFT4.gif)

### Desired behavior:

Ensure the actual element targeted was the one that was clicked. This seems like something that should be added to the actionability, https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Actionability

### Steps to reproduce: (app code and test code)

See https://github.com/MadLittleMods/cypress-test-tiny/pull/2

The test fails with `cypress open` and you can see it also goes wrong in the video with `cypress run` but the test passes(which is also unexpected).

`e2e-specs.js`
```js
describe("page", () => {
  it("shows something", async function() {
    cy.visit(`./test.html`);

    cy.get(".click-me").click();

    cy.get(".click-me").contains("click me [am-i-clicked? true]");
  });
});
```

`test.html`
```html
<textarea class="abc-textarea" autofocus>abc</textarea>
<div class="click-me">
  click me [am-i-clicked? false]
</div>

<script>
  const textarea = document.querySelector(".abc-textarea");
  textarea.addEventListener("blur", () => {
    console.log("textarea blurred");
    textarea.style.height = "200px";
  });

  const elementToClick = document.querySelector(".click-me");
  elementToClick.addEventListener("click", () => {
    console.log("clicked");
    elementToClick.innerText = "click me [am-i-clicked? true]";
    elementToClick.classList.toggle("is-clicked");
  });
</script>
```

### Versions

 - `cypress@3.6.1`
 - Node.js v10.15.1
 - Windows 10

**Comment by jylertones** [comment]
I am experiencing this too - it appears to work on 3.5.0 and break on 3.6.1

**Comment by bimimicah** [comment]
We upgraded from v3.4.1 to v3.6.1 and encountered this same issue with radio button .check() - blur() on the previous selection shifts the content up, and the check() succeeds without actually checking the radio button.

**Comment by jennifer-shehane** [comment]
I'm able to recreate this from the code provided in Electron only. 

However, this was fixed in [4.2.0](https://on.cypress.io/changelog#4-2-0), maybe due to Electron upgrade? https://github.com/cypress-io/cypress/pull/6555. Closing as resolved. 

If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.
